PCF-ADR-0002-release-story-and-branch-maintenance.md

```diff
@@ -0,0 +1,140 @@
+  branch for the minor release. (explained below)
+
+### Rationale
+
+- It is simple and easy to understand and follow. There is no need to maintain multiple long-lived branches or deal with
+  complex merge conflicts.
+- It supports continuous integration and delivery. Every commit on the `main` branch is tested and deployable, ensuring
+  high quality and fast feedback.
+- It encourages collaboration and code review. Pull requests are the central place for discussing and reviewing code
+  changes, ensuring that every change is approved by at least one other developer before merging.
+- It allows us to release frequently and incrementally. We can deploy new features and bug fixes as soon as they are
+  ready, without waiting for a predefined release cycle or schedule.
+
+### How to pull in bug fixes into a release branch
+
+Sometimes, we may need to create a release branch from the `main` branch to prepare for a major or minor release. For
+example, we may want to freeze the feature set for a certain version and only allow bug fixes on that branch. In that
+case, we can use the following steps to pull in bug fixes from the `main` branch into the release branch:
+
+- Create a release branch from the `main` branch with a name that indicates the minor version number, such as `v1.2`.
+- Continue developing new features and bug fixes on the `main` branch as usual.
+- If a bug fix on the `main` branch needs to be included in the release branch, cherry-pick the commit that contains the
+  fix onto the release branch.
+- When the release branch is ready for deployment, create a tag on GitHub with the patch version incremented the, such
+  as `v1.2.1`. This will trigger the deployment pipeline for that tag.
+
+### Release process
+
+The release process for our project is very simple and consists of creating a version tag on GitHub. For example, if we
+want to release version 1.3 of our software, we just need to create a tag named `v1.3` on GitHub from the latest commit
+on the `main` branch. This will trigger the build and deployment pipelines, or someone will manually take the required
+steps for generating build artifacts, etc..
+
+We do not need to create major version tags unless there is a significant change in the software architecture or
+functionality that requires a major version bump.
+
+### Versioning Scheme
+
+All projects should use semantic versioning scheme as it gives the most warning for potentially breaking changes when
+compared to other schemes such as calendar versioning.
+
+## Non Goals
+
+- Dictating how to build code for release (prefer automation, but a manual process is also okay)
+- Setting a release cycle for the codeflare project.
+
+## Open questions
+
+- What artifacts should be built to go alongside the releases? For example, should we generate documentation, binaries,
+  packages, etc.?
+- How should we communicate our releases to our stakeholders and customers? Should we use changelogs, release notes,
```

Comment:
Is there a CodeFlare mail list? At a minimum, we can share these out to the slack channel

Reply:
I'll start a thread in slack too.